pkg/operator/ceph/csi/peermap/clientProfile_test.go

```diff
@@ -0,0 +1,99 @@
+		Executor:  mockExecutor,
+		Clientset: test.New(t, 3),
+	}
+
+	// create fake secret with "peer1" cluster token
+	_, err := fakeContext.Clientset.CoreV1().Secrets(ns).Create(context.TODO(), &peer1Secret, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	clientProfile, err := generateClientProfileMappingsCR(fakeContext, clusterInfo, &fakeSinglePeerCephBlockPool)
+	assert.NoError(t, err)
+
+	assert.Equal(t, 1, len(clientProfile.Spec.BlockPoolMapping))
+	assert.Equal(t, "rook-ceph-primary", clientProfile.Spec.BlockPoolMapping[0].Local.ClientProfileName)
+	assert.Equal(t, "peer1", clientProfile.Spec.BlockPoolMapping[0].Remote.ClientProfileName)
+	assert.Equal(t, 1, clientProfile.Spec.BlockPoolMapping[0].Local.PoolId)
+	assert.Equal(t, 2, clientProfile.Spec.BlockPoolMapping[0].Remote.PoolId)
+}
+
+func TestGenerateClientProfileMappingsWithMultiplePeers(t *testing.T) {
+	clusterInfo := cephclient.AdminTestClusterInfo(ns)
+	fakeContext := &clusterd.Context{
+		Executor:  mockExecutor,
+		Clientset: test.New(t, 3),
+	}
+
+	// create fake secret with "peer1" cluster token
+	_, err := fakeContext.Clientset.CoreV1().Secrets(ns).Create(context.TODO(), &peer1Secret, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	// create fake secret with "peer2" cluster token
+	_, err = fakeContext.Clientset.CoreV1().Secrets(ns).Create(context.TODO(), &peer2Secret, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	// create fake secret with "peer3" cluster token
+	_, err = fakeContext.Clientset.CoreV1().Secrets(ns).Create(context.TODO(), &peer3Secret, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	clientProfile, err := generateClientProfileMappingsCR(fakeContext, clusterInfo, &fakeMultiPeerCephBlockPool)
+	assert.NoError(t, err)
+
+	assert.Equal(t, 3, len(clientProfile.Spec.BlockPoolMapping))
+	assert.Equal(t, "rook-ceph-primary", clientProfile.Spec.BlockPoolMapping[0].Local.ClientProfileName)
+	assert.Equal(t, "peer1", clientProfile.Spec.BlockPoolMapping[0].Remote.ClientProfileName)
+	assert.Equal(t, 1, clientProfile.Spec.BlockPoolMapping[0].Local.PoolId)
+	assert.Equal(t, 2, clientProfile.Spec.BlockPoolMapping[0].Remote.PoolId)
+
+	assert.Equal(t, "rook-ceph-primary", clientProfile.Spec.BlockPoolMapping[1].Local.ClientProfileName)
+	assert.Equal(t, "peer2", clientProfile.Spec.BlockPoolMapping[1].Remote.ClientProfileName)
+	assert.Equal(t, 1, clientProfile.Spec.BlockPoolMapping[1].Local.PoolId)
+	assert.Equal(t, 3, clientProfile.Spec.BlockPoolMapping[1].Remote.PoolId)
+
+	assert.Equal(t, "rook-ceph-primary", clientProfile.Spec.BlockPoolMapping[2].Local.ClientProfileName)
+	assert.Equal(t, "peer3", clientProfile.Spec.BlockPoolMapping[2].Remote.ClientProfileName)
+	assert.Equal(t, 1, clientProfile.Spec.BlockPoolMapping[2].Local.PoolId)
+	assert.Equal(t, 4, clientProfile.Spec.BlockPoolMapping[2].Remote.PoolId)
+}
+
+func TestGetFirstKeyValueAsInt(t *testing.T) {
+	actualKey, actualValue, err := getFirstKeyValueAsInt(map[string]string{"3": "5"})
```

Comment:
Related to the discussion above on the ordering, why don't we need a deterministic result? If there are multiple entries you will get different results for the first item. For example, if you add multiple entries to the map, this test will fail sometimes.
```suggestion
	actualKey, actualValue, err := getFirstKeyValueAsInt(map[string]string{"3": "5", "6": "7"})
```